test/integration/test_konnect_extension.go

```diff
@@ -39,40 +51,331 @@ func TestKonnectExtensionKonnectGatewayControlPlaneNamespacedRef(t *testing.T) {
+					Name: lo.ToPtr("httpbin"),
+					URL:  lo.ToPtr(fmt.Sprintf("http://%s.%s.svc.cluster.local/", service.Name, ns.Name)),
+					Host: fmt.Sprintf("%s.%s.svc.cluster.local", service.Name, ns.Name),
+				}
+			},
+		)
+		t.Logf("Waiting for KongService to be updated with Konnect ID")
+		require.EventuallyWithT(t, func(t *assert.CollectT) {
+			err := GetClients().MgrClient.Get(GetCtx(), k8stypes.NamespacedName{Name: ks.Name, Namespace: ks.Namespace}, ks)
+			require.NoError(t, err)
+			assertKonnectEntityProgrammed(t, ks)
+		}, testutils.ObjectUpdateTimeout, testutils.ObjectUpdateTick)
+		t.Cleanup(deleteObjectAndWaitForDeletionFn(t, ks))
+
+		t.Log("Creating a KongRoute to the service")
+		kr := deploy.KongRouteAttachedToService(t, ctx, clientNamespaced, ks,
+			func(obj client.Object) {
+				s := obj.(*configurationv1alpha1.KongRoute)
+				s.Spec.KongRouteAPISpec.Paths = []string{"/test"}
+			},
+		)
+		t.Logf("Waiting for KongRoute to be updated with Konnect ID")
+		require.EventuallyWithT(t, func(t *assert.CollectT) {
+			err := GetClients().MgrClient.Get(GetCtx(), k8stypes.NamespacedName{Name: kr.Name, Namespace: kr.Namespace}, kr)
+			require.NoError(t, err)
+
+			assertKonnectEntityProgrammed(t, kr)
+		}, testutils.ObjectUpdateTimeout, testutils.ObjectUpdateTick)
+		t.Cleanup(deleteObjectAndWaitForDeletionFn(t, kr))
+
+		t.Run("KonnectExtension with KonnectID control plane ref", func(t *testing.T) {
+			t.Run("manual secret provisioning", func(t *testing.T) {
+				t.Logf("Creating a Secret Certificate for the KonnectExtension")
+				secretCert := deploy.Secret(
+					t, ctx, clientNamespaced,
+					map[string][]byte{
+						consts.TLSCRT: cert,
+						consts.TLSKey: key,
+					},
+					deploy.WithLabel("konghq.com/konnect-dp-cert", "true"),
+				)
+				t.Cleanup(deleteObjectAndWaitForDeletionFn(t, secretCert.DeepCopy()))
 
-	t.Cleanup(deleteObjectAndWaitForDeletionFn(t, cp.DeepCopy()))
+				t.Logf("Creating a KonnectExtension")
+				keWithKonnectIDCPRef := deploy.KonnectExtension(
+					t, ctx, clientNamespaced,
+					deploy.WithKonnectConfiguration[*konnectv1alpha1.KonnectExtension](konnectv1alpha1.KonnectConfiguration{
+						APIAuthConfigurationRef: konnectv1alpha1.KonnectAPIAuthConfigurationRef{
+							Name: authCfg.Name,
+						},
+					}),
+					deploy.WithKonnectIDControlPlaneRef(cp),
+					setKonnectExtensionDPCertSecretRef(t, secretCert),
+				)
+				t.Cleanup(deleteObjectAndWaitForDeletionFn(t, keWithKonnectIDCPRef.DeepCopy()))
+
+				params := KonnectExtensionTestBodyParams{
+					konnectControlPlane: cp,
+					konnectExtension:    keWithKonnectIDCPRef,
+					secret:              secretCert,
+					client:              clientNamespaced,
+					authConfigName:      authCfg.Name,
+					namespace:           ns.Name,
+				}
+				konnectExtensionTestBody(t, params)
+			})
+
+			t.Run("automatic secret provisioning", func(t *testing.T) {
+				t.Logf("Creating a KonnectExtension")
+				keWithKonnectIDCPRef := deploy.KonnectExtension(
```

Comment:
```suggestion
				keWithKonnectIDCPRef := deploy.KonnectExtension(
```

Nit: `deploy` funcs do log about the created object: https://github.com/Kong/gateway-operator/blob/624554aa7ed45a31d973afa59d6321640153181d/test/helpers/deploy/deploy_resources.go#L1032